tripy/tripy/frontend/trace/ops/binary_elementwise.py

```diff
@@ -183,7 +183,10 @@ def to_flat_ir(self, inputs, outputs):
     dtype_constraints={"other": "T1", constraints.RETURN_VALUE: "T1"},
     function_name="__radd__",
 )
-def __add__(self: Union["tripy.Tensor", Any], other: Union["tripy.Tensor", Any]) -> "tripy.Tensor":
+def __add__(
+    self: Union["tripy.Tensor", "tripy.types.tensor_literal"],
```

Comment:
Is it worth having a type variable for this? I expect it'll be fairly common

Reply:
Good idea. There were nearly 30 instances, so I added a variable `ConvertibleToTensor`. Not 100% sure on the name.

Reply:
`ConvertibleToTensor` seems fine. Another option is `TensorLike` (similar to `ArrayLike` in numpy).

Reply:
I like `TensorLike`, I'll go with that.